integration/setup/setup.go

```diff
@@ -342,3 +327,68 @@ func insertBenchmarkProvider(tb testing.TB, ctx context.Context, collection *mon
 
 	return
 }
+
+// cleanupCollection cleans up the database if `opts.DatabaseName` is empty,
+// otherwise it deletes all documents in the collection for `ferretdb-yugabytedb` backend
+// or drops the collection for other backends. The drop collection error is optionally ignored.
+func cleanupCollection(tb testing.TB, ctx context.Context, database *mongo.Database, collection *mongo.Collection, opts *SetupOpts, ignoreErr bool) {
+	tb.Helper()
+
+	if opts.DatabaseName == "" {
+		cleanupDatabase(tb, ctx, database)
+
+		return
+	}
+
+	if IsYugabyteDB(tb) {
+		// dropping collection or database fails for ferretdb-yugabytedb
+		// TODO https://github.com/yugabyte/yugabyte-db/issues/27698
+		_, err := collection.DeleteMany(ctx, bson.D{})
+		require.NoError(tb, err)
+
+		return
+	}
+
+	err := collection.Drop(ctx)
+
+	if ignoreErr {
+		return
+	}
+
+	require.NoError(tb, err)
+}
+
+// cleanupDatabase drops all users, then deletes all collections for `ferretdb-yugabytedb` backend
+// or drops the database for other backends.
+func cleanupDatabase(tb testing.TB, ctx context.Context, database *mongo.Database) {
+	err := database.RunCommand(ctx, bson.D{{"dropAllUsersFromDatabase", 1}}).Err()
+	require.NoError(tb, err)
+
+	if IsYugabyteDB(tb) {
+		// dropping collection or database fails for ferretdb-yugabytedb
+		// TODO https://github.com/yugabyte/yugabyte-db/issues/27698
+		var cursor *mongo.Cursor
+		cursor, err = database.ListCollections(ctx, bson.D{})
+		require.NoError(tb, err)
+
+		type collectionInfo struct {
+			Name string `bson:"name"`
+		}
+
+		var res []collectionInfo
+
+		err = cursor.All(ctx, &res)
+		require.NoError(tb, cursor.Close(ctx))
+		require.NoError(tb, err)
```

Comment:
We probably could use https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#Database.ListCollectionNames